src/Symfony/Component/Validator/Constraint.php

```diff
@@ -231,10 +232,16 @@ public function __get($option)
      * @param string $group
      *
      * @api
+     *
+     * @throws GroupDefinitionException If the implicit group has been explicitly configured in current constraint.
      */
     public function addImplicitGroupName($group)
     {
-        if (in_array(self::DEFAULT_GROUP, $this->groups) && !in_array($group, $this->groups)) {
+        if (in_array($group, $this->groups)) {
+            throw new GroupDefinitionException(sprintf('The implicit group "%s" has already been explicitly configured in constraint %s', $group, get_class($this)));
```

Comment:
Thanks for your feedback. Since I'm not an English native speaker, could you advice the rewording?